src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -463,6 +471,106 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }))
   }
 
+  const prepareVoteAuxiliaryData = (
+    hwStakeSigningFile: HwSigningData,
+    votingPublicKeyHex: string,
+    addressParameters: _AddressParameters,
+    nonce: BigInt,
+  ): LedgerTypes.TxAuxiliaryData => ({
+    type: LedgerTypes.TxAuxiliaryDataType.TUPLE,
+    params: {
+      metadata: {
+        type: LedgerTypes.TxMetadataType.CATALYST_REGISTRATION,
+        params: {
+          votingPublicKeyHex,
+          stakingPath: hwStakeSigningFile.path,
+          rewardsDestination: {
+            type: addressParameters.addressType,
+            params: {
+              spendingPath: addressParameters.paymentPath,
+              stakingPath: addressParameters.stakePath,
+            },
+          },
+          nonce: `${nonce}`,
+        },
+      },
+    },
+  })
+
+  const prepareDummyInput = (): LedgerTypes.TxInput => ({
+    txHashHex: '0'.repeat(64),
+    outputIndex: 0,
+    path: parseBIP32Path('1852H/1815H/0H/0/0'),
+  })
+
+  const prepareDummyOutput = (): LedgerTypes.TxOutput => ({
+    destination: {
+      type: LedgerTypes.TxOutputDestinationType.DEVICE_OWNED,
+      params: {
+        type: LedgerTypes.AddressType.BASE,
+        params: {
+          spendingPath: parseBIP32Path('1852H/1815H/0H/0/0'),
+          stakingPath: parseBIP32Path('1852H/1815H/0H/2/0'),
+        },
+      },
+    },
+    amount: '1',
+  })
+
+  const prepareDummyTx = (
+    network: LedgerTypes.Network, auxiliaryData: LedgerTypes.TxAuxiliaryData,
+  ): LedgerTypes.SignTransactionRequest => ({
+    signingMode: LedgerTypes.TransactionSigningMode.ORDINARY_TRANSACTION,
+    tx: {
+      network,
+      inputs: [prepareDummyInput()],
+      outputs: [prepareDummyOutput()],
+      fee: 1,
+      ttl: 1,
```

Comment:
once you update to latest ledgerjs pre-release https://github.com/vacuumlabs/ledgerjs-cardano-shelley/releases/tag/v3.1.0-catalyst-rc.2 you should be able to set ttl to zero

Reply:
I updated the lib, but I'm still getting `Error: invalid ttl`

Reply:
ok, noted, will debug it

Reply:
apparently the validation of uints incorrectly rejected all zero values, it's fixed in this pre-release: https://github.com/vacuumlabs/ledgerjs-cardano-shelley/releases/tag/v3.1.0-catalyst-rc.3

Reply:
also please check if this earlier, possibly related, bug isnt already fixed https://github.com/vacuumlabs/cardano-hw-cli/issues/60

Reply:
I'm still getting same error.